hack/verify-golint.sh

```diff
@@ -10,45 +11,21 @@ if ! which golint &>/dev/null; then
 fi
 
 GO_VERSION=($(go version))
-echo "Detected go version: $(go version)"
 
-if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.4|go1.5') ]]; then
-  echo "Unknown go version '${GO_VERSION}', skipping golint."
+if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.[4-6]') ]]; then
+  echo "Unknown go version '${GO_VERSION[@]:2}', skipping golint."
   exit 0
 fi
 
 STI_ROOT=$(dirname "${BASH_SOURCE}")/..
```

Comment:
That's a good improvement, but touches 49 lines in 14 files... I fear it will just make this PR fuzzier to understand.
I will not do it here because there's a risk of things outside of this repo dependent on the name.
And if we change STI_ROOT, then we can go ahead and change all STI_* variables... but that's really a riskier move.